packages/react-native/Libraries/Text/TextInput/Multiline/RCTUITextView.m

```diff
@@ -298,7 +298,10 @@ - (CGRect)caretRectForPosition:(UITextPosition *)position
     return CGRectZero;
   }
 
-  return [super caretRectForPosition:position];
+  CGRect originalRect = [super caretRectForPosition:position];
+  UIFont *font = self.font;
+  originalRect.size.height = font.pointSize - font.descender;
+  return CGRectMake(originalRect.origin.x, originalRect.origin.y - font.descender , originalRect.size.width, originalRect.size.height);
```

Comment:
You should either mutate the rect here, or create a new one. Combining both is not necessary
```suggestion
  CGRect rect = [super caretRectForPosition:position];
  UIFont *font = self.font;
  rect.size.height = font.pointSize - font.descender;
  rect.origin.y -= font.descender;
  return rect;
```